tmt/steps/__init__.py

```diff
@@ -2096,6 +2101,144 @@ class Plugin(BasePlugin[StepDataT, PluginReturnValueT]):
     Common parent of all step plugins that do work against a particular guest
     """
 
+    @overload
+    def write_command_report(
+        self,
+        *,
+        path: Path,
+        label: str,
+        timer: Stopwatch,
+        command: Optional[Union[Command, ShellScript]] = None,
+        output: CommandOutput,
+        exc: None = None,
+    ) -> None:
+        pass
+
+    @overload
+    def write_command_report(
+        self,
+        *,
+        path: Path,
+        label: str,
+        timer: Stopwatch,
+        command: Optional[Union[Command, ShellScript]] = None,
+        output: None = None,
+        exc: Exception,
+    ) -> None:
+        pass
+
+    def write_command_report(
+        self,
+        *,
+        path: Path,
+        label: str,
+        timer: Stopwatch,
+        command: Optional[Union[Command, ShellScript]] = None,
+        output: Optional[CommandOutput] = None,
+        exc: Optional[Exception] = None,
+    ) -> None:
+        """
+        Create a report for a given command outcome.
+
+        Combines :py:func:`tmt.utils.render_report` and
+        :py:func:`tmt.utils.render_command_report` to create a report
+        for a given command, its output, and possibly its failure. The
+        report is then written into a file that is created if it did not
+        exist, or added at the end of of the file if it did existed
+        already.
```

Comment:
```suggestion
        report is then appended to the report file (``path``).
```
Seems excessive to mention that the file is created when you append to it. Was looking if we could reference the parameters, but apparently not :/